### PR TITLE
#221 sanitizers errors should break unit tests

### LINKED ISSUE #221: Errors found by sanitizers don't break unit tests
Which makes that errors semi-invisible. Possible fix -> https://github.com/sfc-gh-ebrossard/googletest/blob/master/googletest/docs/advanced.md#sanitizer-integration

### PR DESCRIPTION
**Pull request by jstrzebonski**
Fixes #221

### CODE REVIEW

File: tests/unit/test_footprinter.cc [hunk 1]
```diff
@@ -177,15 +177,6 @@ TEST_F(TestFootprinter, test_fundamental_types) {
...
   }
-
-  {
-    int arr[] = {1, 2, 3, 4, 5};
-    EXPECT_EQ(
-      checkpoint::getMemoryFootprint(arr),
-      sizeof(arr)
-      // actually: sizeof(&arr) + sizeof(arr)
-    );
-  }
```
Comment (jstrzebonski) [inline tests/unit/test_footprinter.cc:188]: UBSAN found an error here:
```
reference binding to misaligned address 0x000200000001 for type 'int', which requires 4 byte alignment
```


`dispatch::with()` does `auto val = cleanType(&target);` which eventually does `std::decay` on C-array type. Simplifying - it casts it to a raw pointer, making serialization of an array not possible. That's why I'm removing this test.

File: tests/unit/main.cc [hunk 2]
```diff
@@ -54,6 +54,16 @@
 #include <Kokkos_Core.hpp>
 #endif
 
+extern "C" {
+void __ubsan_on_report() {
+   FAIL() << "Encountered an undefined behavior sanitizer error";
+}
+
+void __asan_on_error() {
+  FAIL() << "Encountered an address sanitizer error";
```
Comment (cz4rs) [inline tests/unit/main.cc:63]: can you post an example of how does such error look in the CI? (is the complete output from ASAN included?)

**Review by cz4rs — APPROVED** [review]
looks good to me!

**Review by JacobDomagala — APPROVED** [review]

File: tests/unit/main.cc (hunk 2 above)
Comment (jstrzebonski) [reply tests/unit/main.cc:63]: Yes. Take a look -> https://github.com/DARMA-tasking/checkpoint/pull/222/checks?check_run_id=3115530025#step:7:493

image

**Review by lifflander — APPROVED** [review]

**Pull request merged**